src/Connector.SqlServer/Connector/SqlServerConnector.cs

```diff
@@ -379,10 +396,18 @@ async Task renameTable(string currentName, string updatedName, string context)
             string id)
         {
             var config = await base.GetAuthenticationDetails(executionContext, providerDefinitionId);
+            var schema = config.GetSchema();
+            var container = new Container(id, StreamMode);
+            var tasks = new List<Task> { removeTable(container.PrimaryTable, "Data") };
+            foreach (var table in container.Tables)
+                if (await CheckTableExists(executionContext, providerDefinitionId, table.Value.Name.GetValue()))
+                    tasks.Add(removeTable(table.Value.Name, table.Key));
 
-            async Task removeTable(string name, string context)
+            await Task.WhenAll(tasks);
+
+            async Task removeTable(SanitizedSqlName name, string context)
```

Comment:
Agree, same as above
"agree about separate internal methods or even just a straightforward method body instead of local functions. Unfortunately haven't had time to properly refactor this.
At least not in scope of this task."